lib/bugsnag/rake.rb

```diff
@@ -2,36 +2,23 @@
 
 Rake::TaskManager.record_task_metadata = true
 
-module Bugsnag::Rake
-  def self.included(base)
-    base.extend ClassMethods
-    base.class_eval do
-      class << self
-        alias_method :original_define_task, :define_task
-        alias_method :define_task, :bugsnag_define_task
-      end
-    end
-  end
+class Rake::Task
+
+  def execute_with_bugsnag(args=nil)
+    old_task = Bugsnag.configuration.request_data[:bugsnag_running_task]
+    Bugsnag.set_request_data :bugsnag_running_task, self
 
-  module ClassMethods
-    def bugsnag_define_task(*args, &block)
-      task = self.original_define_task(*args) do |*block_args|
-        begin
-          old_task = Bugsnag.configuration.request_data[:bugsnag_running_task]
-          Bugsnag.set_request_data :bugsnag_running_task, task
+    execute_without_bugsnag(args)
 
-          yield(*block_args) if block_given?
-        rescue Exception => e
-          Bugsnag.auto_notify(e)
-          raise
-        ensure
-          Bugsnag.set_request_data :bugsnag_running_task, old_task
-        end
-      end
-    end
+  rescue Exception => ex
+    Bugsnag.auto_notify(ex)
+    raise
+  ensure
+    Bugsnag.set_request_data :bugsnag_running_task, old_task
```

Comment:
Rake tasks can call each other so this ensures that exceptions passed to Bugsnag.notify will have the correct task meta-data.